apis/v1alpha1/validation/validation.go

```diff
@@ -75,3 +75,40 @@ func validateListenerHostname(listeners []gatewayv1a1.Listener, path *field.Path
 	}
 	return errs
 }
+
+// ValidateHTTPRoute validates route according to the Gateway API specification.
+// For additional details of the HTTPRoute spec, refer to:
+//  https://gateway-api.sigs.k8s.io/spec/#networking.x-k8s.io/v1alpha1.HTTPRoute
+func ValidateHTTPRoute(route *gatewayv1a1.HTTPRoute) field.ErrorList {
+	return validateHTTPRouteSpec(&route.Spec, field.NewPath("spec"))
+}
+
+// validateHTTPRouteSpec validates whether required fields of spec are set according to the
+// HTTPRoute specification.
+func validateHTTPRouteSpec(spec *gatewayv1a1.HTTPRouteSpec, path *field.Path) field.ErrorList {
+	return validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules"))
+}
+
+// validateHTTPRouteUniqueFilters validates whether each core and extended filter
+// is used at most once in each rule.
+func validateHTTPRouteUniqueFilters(rules []gatewayv1a1.HTTPRouteRule, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+
+	for i, rule := range rules {
+		counts := map[gatewayv1a1.HTTPRouteFilterType]int{}
+		for _, filter := range rule.Filters {
+			counts[filter.Type]++
+		}
+		// custom filters don't have any validation
+		counts[gatewayv1a1.HTTPRouteFilterExtensionRef] = 0
```

Comment:
I think it might be better to have a specific list of filters that can or can't be repeated. Maybe declare a variable at the top of this file that lists the kinds of filters that can be repeated? Something similar to the [upstream networking validation](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/networking/validation/validation.go#L56).